core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableFilter.java

```diff
@@ -68,4 +73,23 @@ public Result implement(EnumerableRelImplementor implementor, Prefer pref) {
     // EnumerableCalc is always better
     throw new UnsupportedOperationException();
   }
+
+  @Override public Pair<RelTraitSet, List<RelTraitSet>> passThroughTraits(
+      RelTraitSet required) {
+    if (required.getCollation() == null) {
```

Comment:
Indeed, the collation check is needed. Let's add this check back
```
required.getCollation() == RelCollations.EMPTY
```
Because it is possible that only convention is different.

Reply:
Or you can simply try this
```
if (required.isDefaultSansConvention()) {
  return null;
}
```

Reply:
In fact we still need `collation == null`. Turns out that after turning on globally top-down opt, there still NPE hitting this `if` because collation is null.